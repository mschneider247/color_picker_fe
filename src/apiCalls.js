export const fetchAllProjects = async () => {
	const response = await fetch('https://color-picker-be.herokuapp.com/api/v1/projects');
	const data = await response.json();
	return data;
};

export const fetchAllPalettes = async () => {
	const response = await fetch('https://color-picker-be.herokuapp.com/api/v1/palettes');
	const data = await response.json();
	return data;
};

export const fetchProject = async (id) => {
	const response = await fetch(`https://color-picker-be.herokuapp.com/api/v1/projects/${id}`);
	const data = await response.json();
	return data;
};

export const fetchPalette = async (id) => {
	const response = await fetch(`https://color-picker-be.herokuapp.com/api/v1/palettes/${id}`);
	const data = await response.json();
	return data;
};

export const addProject = async (info) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(info),
    headers: {
      'Content-Type': 'application/json'
    }
  };

  const resp = await fetch('https://color-picker-be.herokuapp.com/api/v1/projects', options)
  const data = await resp.json();
  return data
};

export const addPalette = async (info) => {
  const options = {
    method: 'POST',
    body: JSON.stringify(info),
    headers: {
      'Content-Type': 'application/json'
    }
  };

  const resp = await fetch('https://color-picker-be.herokuapp.com/api/v1/palettes', options)
  const data = await resp.json();
  return data
};

export const deleteProject = async (id) => {
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  };

  const resp =  await fetch(`https://color-picker-be.herokuapp.com/api/v1/projects/${id}`, options);
  const data = await resp.json();
  return data
};

export const deletePalette = async (id) => {
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  };

  const resp =  await fetch(`https://color-picker-be.herokuapp.com/api/v1/palettes/${id}`, options);
  const data = await resp.json();
  return data
};

export const patchProjectName = async (name, id) => {
  const options = {
    method: 'PATCH',
    body: JSON.stringify(name),
    headers: {
      'Content-Type': 'application/json'
    }
  };

  const resp = await fetch(`https://color-picker-be.herokuapp.com/api/v1/projects/${id}`, options)
  const data = await resp.json();
  return data
};