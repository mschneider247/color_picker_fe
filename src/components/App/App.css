#app {
  width: 100%;
  height: 1000px;
  padding: 5%;
  background-image: url('../../Images/FINAL_color_picker_FE.jpg');
  background-size: cover;
}