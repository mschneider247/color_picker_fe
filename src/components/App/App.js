import React, { Component } from 'react';
import './App.css';
import Welcome from '../Welcome/Welcome';
import Projects from '../Projects/Projects'
import { fetchAllProjects, fetchAllPalettes } from '../../apiCalls';

class App extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      palettes: [],
      projectName: '',
      paletteName: '',
      projectId: 0,
      id: 0,
      color1: '',
      color2: '',
      color3: '',
      color4: '',
      color5: ''
    }
  }

  componentDidMount = () => {
    fetchAllProjects()
      .then(projects => this.setState({ projects }))
      .catch(error => console.log(error))

    fetchAllPalettes()
      .then(palettes => this.setState({ palettes }))
      .catch(error => console.log(error))
  }

  updatePalettes = (e) => {
    this.setState({ projectName: e.target.innerText });
    this.setState({ projectId: parseInt(e.target.value) });
  }

  render() {
    const displayPalettes = this.state.palettes.filter(palette => palette.projectId === this.state.projectId)
    const paletteName = displayPalettes.map(palette => palette.name)
    const displayProjects = this.state.projects.map(project => {
        return (<button value={project.projectId} onClick={(e) => this.updatePalettes(e)}>{project.name}</button>)
      })
    return (
      <section>
        <Welcome />
        <Projects projects={displayProjects} />
        {/* {displayProjects} */}
        {paletteName}
      </section>
    );
  };
}

export default App;
