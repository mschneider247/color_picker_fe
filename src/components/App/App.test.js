import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { fetchAllProjects, fetchAllPalettes, fetchProject, fetchPalette, addProject, addPalette, deleteProject, deletePalette, patchProjectName, patchPaletteColor } from '../../apiCalls';

jest.mock('../../apiCalls');

describe('App component', () => {
  let wrapper;
  let mockProjects = [{
  	id: 20,
    projectId: 1,
    name: 'BYOB',
  }];
  let mockProject = {
  	id: 20,
    projectId: 1,
    name: 'BYOB',
  };
  let mockPalettes = [{
  	id: 5,
    projectId: 1,
    name: 'cool colors',
    color1: '#668B8B',
    color2: '#B4CDCD',
    color3: '#2F4F4F',
    color4: '#7A8B8B',
    color5: '#838B8B',
  }];
  let mockPalette = {
  	id: 5,
    projectId: 1,
    name: 'cool colors',
    color1: '#668B8B',
    color2: '#B4CDCD',
    color3: '#2F4F4F',
    color4: '#7A8B8B',
    color5: '#838B8B',
  };
  let mockEvent = { target: { value: 10, innerText: 'Title'}};
  let mockIndex = 3;
  let mockId = { id: 10 };
  let mockResponse = `Project ${mockProject.id} deleted`

  beforeEach(() => {
    wrapper = shallow(<App/>)
  });

  fetchAllProjects.mockImplementation(() => {
  	return Promise.resolve(mockProjects)
  });

  fetchAllPalettes.mockImplementation(() => {
  	return Promise.resolve(mockPalettes)
  });

  addProject.mockImplementation(() => {
  	return Promise.resolve(mockId)
  });

  deleteProject.mockImplementation(() => {
  	return Promise.resolve(mockResponse)
  });

  deletePalette.mockImplementation(() => {
  	return Promise.resolve(mockResponse)
  });

  it('App should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchAllProjects when getProjects is invoked', () => {
  	wrapper.instance().getProjects();
  	expect(fetchAllProjects).toHaveBeenCalled();
  });

  it('should update state when getProjects is invoked', () => {
  	wrapper.instance().getProjects();
  	expect(wrapper.state('projects')).toEqual(mockProjects);
  });

  it('should call fetchAllPalettes when getPalettes is invoked', () => {
  	wrapper.instance().getPalettes();
  	expect(fetchAllPalettes).toHaveBeenCalled();
  });

  it('should update state when getPalettes is invoked', () => {
  	wrapper.instance().getPalettes();
  	expect(wrapper.state('palettes')).toEqual(mockPalettes);
  });

  it('should update state when updateProject is invoked', () => {
  	wrapper.instance().updateProject(mockEvent);
  	expect(wrapper.state('projectName')).toEqual('Title');
  	expect(wrapper.state('projectId')).toEqual(10);
  });

  it('should update state when updatePalette is invoked', () => {
  	wrapper.instance().updatePalette(mockEvent, mockPalette);
  	expect(wrapper.state('paletteName')).toEqual('Title');
  	expect(wrapper.state('paletteId')).toEqual(10);
  	expect(wrapper.state('paletteId')).toEqual(10);
  });

  it('should update state when updateLockedIndex is invoked', () => {
  	wrapper.instance().updateLockedIndex(mockIndex);
  	expect(wrapper.state('lockedIndex')).toEqual(mockIndex);
  });

  it('should update state when updatePaletteName is invoked', () => {
  	let mockName = 'Name';
  	wrapper.instance().updatePaletteName(mockName);
  	expect(wrapper.state('paletteName')).toEqual(mockName);
  });

  it('should invoke addProject when addNewProject is called', async () => {
  	let mockRequest = { projectId: 2, name: 'Name' }
  	wrapper.instance().addNewProject(mockRequest.name);
  	await expect(addProject).toHaveBeenCalledWith(mockRequest);
  });

  it('should invoke deletePalette when removePalette is called', async () => {
  	mockId = 4;
  	mockResponse = `Palette ${mockId} deleted`
  	wrapper.instance().removePalette(mockId);
  	await expect(deletePalette).toHaveBeenCalledWith(mockId);
  });
});
