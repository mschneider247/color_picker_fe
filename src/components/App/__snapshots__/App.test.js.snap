// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App should match its snapshot 1`] = `ShallowWrapper {}`;
